### DISCUSSION TITLE
GroupBy or SequenceEqual method's comparer as Lambda expression

### DISCUSSION THREAD

**Discussion by NagayamaToshiaki** [dotnet/csharplang#6110] [Language Ideas]
It's quite tedious to make a class that imprements IEqualityComparer each time you need to compare custom classes. Those class would often need only once. Can it be replaced with Func<TKey, TKey, bool>?

**Comment by TahirAhmadov** [comment]
https://github.com/dotnet/runtime/issues/

**Reply by jnm2** [reply]
Also consider upvoting https://github.com/dotnet/runtime/issues/33877
